src/owl/config/configure.ml

```diff
@@ -76,31 +115,53 @@ let get_ocaml_devmode_flags _c =
   if not enable_devmode then [] else [ "-w"; "-32-27-6-37-3" ]
 
 
-let default_cflags =
-  try
-    Sys.getenv "OWL_CFLAGS"
-    |> String.trim
-    |> String.split_on_char ' '
-    |> List.filter (fun s -> String.trim s <> "")
-  with
-  | Not_found ->
-    [ (* Basic optimisation *)
-      "-g"
-    ; "-O3"
-    ; "-Ofast"
-    ; (* FIXME: experimental switches *)
-      (* "-mavx2"; "-mfma"; "-ffp-contract=fast"; *)
-      (* Experimental switches, -ffast-math may break IEEE754 semantics*)
-      "-march=native"
-    ; "-mfpmath=sse"
-    ; "-funroll-loops"
-    ; "-ffast-math"
-    ; (* Configure Mersenne Twister RNG *)
-      "-DSFMT_MEXP=19937"
-    ; "-msse2"
-    ; "-fno-strict-aliasing"
-    ]
-
+let default_config c =
+  let os =
+    let platform = C.C_define.import c ~includes:[ ] ~prelude:detect_system_os [ "PLATFORM_NAME", String ] in
+    match List.map snd platform with
+    | [ String "android" ] -> `android
+    | [ String "ios" ]     -> `ios
+    | [ String "linux" ]   -> `linux
+    | [ String "mac" ]     -> `mac
+    | [ String "windows" ] -> `windows
+    | _                    -> `unknown
+  in
+  let arch =
+    let arch = C.C_define.import c ~includes:[ ] ~prelude:detect_system_arch [ "PLATFORM_ARCH", String ] in
+    match List.map snd arch with
+    | [ String "x86_64" ] -> `x86_64
+    | [ String "x86" ]    -> `x86
+    | [ String "arm64" ]  -> `arm64
+    | [ String "arm" ]    -> `arm
+    | _                   -> `unknown
+  in
+  let cflags =
+    try
+      Sys.getenv "OWL_CFLAGS"
+      |> String.trim
+      |> String.split_on_char ' '
+      |> List.filter (fun s -> String.trim s <> "")
+    with
+    | Not_found ->
+      [ (* Basic optimisation *) "-g"; "-O3"; "-Ofast" ]
+      @ (match arch, os with
+        | `arm64, `mac -> [ "-mcpu=apple-m1" ]
+        | `x86_64, _   -> [ "-march=native"; "-mfpmath=sse"; "-msse2" ]
+        | _            -> [])
+      @ [ (* Experimental switches, -ffast-math may break IEEE754 semantics*)
+          "-funroll-loops"
+        ; "-ffast-math"
+        ; (* Configure Mersenne Twister RNG *)
+          "-DSFMT_MEXP=19937"
+        ; "-fno-strict-aliasing"
+        ]
+  in
+  (* homebrew M1 issue workaround, works only if users use the default homebrew path *)
+  let libs =
+    let p0 = "/opt/homebrew/opt/gcc/lib/gcc/11/" in
```

Comment:
```suggestion
    let p0 = "/opt/homebrew/opt/gcc/lib/gcc/current/" in
```

Reply:
That works with this patch

Reply:
(as I have gcc12!)

Reply:
This is annoying. It would be nice for them to have a working configuration no matter what the gcc version installed…

Reply:
using `current` works whatever the version of GCC installed as this is supposed to point to one specific version.

Reply:
Ah perfect. Then I’ll amend the PR